en/complete_vehicles/atl_mantis_edu.md

```diff
@@ -0,0 +1,169 @@
+
+To update the vehicle with this firmware: rename the file to **autopilot.px4**, copy it to root of the SD card, and restart the vehicle (as described above).
+
+TODO: support easy upload via WiFi.
+
+## Using MAVSDK with the Mantis
+
+By default MAVSDK connects via MAVLink on UDP port 14540.
+This port is configured for the Mantis, so MAVSDK examples and tests should mostly work out of the box.
+
+TODO: Add more detailed tutorial how to use MAVSDK with the Mantis.
+
+## Connection
+
+You can either connect to the drone using WiFi, or using a USB cable connected to the remote control.
+The latter part is work in progress.
+
+You can also access the SD card and the internal storage by connecting to the drone using a USB-C cable.
+Note that the WiFi will be switched off when USB is connected.
+
+### MAVLink Connection
+
+MAVLink traffic is sent via Wifi on 3 different ports:
+
+1. UDP port 14550: to QGC default auto-connect.
+2. UDP port 14540: to SDK (MAVSDK, MAVROS) default connection.
+3. UDP port 14530: to RC controller.
+
+### HTTP connection
+
+#### Download Photos/Videos from Internal Storage or SD Card
+
+There is an HTTP server running on 192.168.42.1 that you can use to view or download the media files.
+
+#### Upload Files to SD Card
+
+You can upload files to the SD card when connected to the WiFi of the drone.
+
+The command to upload a file is:
+```sh
+curl -F "image=@$HOME/path/to/file.ext" -H "Expect:" http://192.168.42.1/cgi-bin/upload
+```
+
+For bigger (multi-part) upload, use:
+```sh
+curl -F "image=@$HOME/path/to/big-file.ext" -H "Expect:" -H "File-Size: $(stat -c%s $HOME/path/to/big-file.ext)" http://192.168.42.1/cgi-bin/upload
+```
+
+### RTSP Connection
+
+Video is streamed at 1280x720 at 30 Hz over RTSP.
+QGC should connect automatically, as Mantis broadcasts the stream parameters using the [VIDEO_STREAM_INFORMATION](https://mavlink.io/en/messages/common.html#VIDEO_STREAM_INFORMATION) MAVLink message.
+
+If you want to connect manually, you can do so on `rtsp://192.168.42.1:554/live`.
+
+
+## Troubleshooting/FAQ
+
+### MAVLink and the camera don't connect even though WiFi is connected
+
+Try resetting the camera by pressing a paperclip into the small hole above the SD card slot for 5 seconds.
+The camera should reboot and reset some of its internal settings.
+
+### Why does it take over a minute for the WiFi / connection to appear?
+
+The Mantis uses DFS channels on 5 GHz.
+In order to be allowed to use these channels, the wifi module needs to listen on these frequency bands for some time to make sure they are not occupied by radars, and other official users.
+Once it is verified that the bands are unoccopied, the WiFi will start.
+
+TODO: check if this is required in the US
+
+### The WiFi does not come up
+
+If the WiFi does not come up after a minute or so, the most common cause is that USB is connected (you can either connect WiFi or USB, but not both at the same time).
```

Comment:
Note, I moved this below the other one - chances are the WiFi is just slow to come up as per preceding point.

Reply:
Thx.